BookSearch.tsx

```diff
@@ -6,18 +6,33 @@ import books from './book_test'
 
 class BookSearch extends Component {
 
-    getBooks(text) {
-        getBooksFromApiWithText(text).then(data => console.log(data));
+    constructor(props) {
+        super(props)
+        this.state = { books: [] }
+        this.searchedText = ""
+    }
+
+    getBooks() {
+        getBooksFromApiWithText(this.searchedText).then(data => {
+            this.setState({ books: data.docs })
+        })
+    }
+
+    searchTextInputChanged(text){
+        this.searchedText = text;
     }
 
     render() {
         return (
             <View>
                 <StatusBar barStyle="dark-content" />
-                <TextInput onSubmitEditing={(text) => this.getBooks(text)} placeholder="Tester"/>
+                <TextInput 
+                    onSubmitEditing={(text) => this.getBooks()} 
+                    placeholder="Rechercher"
+                    onChangeText={(text) => this.searchTextInputChanged(text) }/>
```

Comment:
Quand tu change une valuer dans ton composant, il vaut mieux le fair eavec un `setState`. Comme ça dès que ta valeur change, si elle est utilisée dans ton rendu, elle sera affichée avec sa dernière mise à jour.
```suggestion
                    onChangeText={(text) => this.setState({searchedText: text}) }/>
```

En fait la fonction setState est propre à React et permet de dire au composant "Attention, une variable a changée de valeur, si un élément a besoin d'elle, alors re-rendre le composant"